tests/test_scan_container.py

```diff
@@ -35,7 +36,13 @@
 ]
 
 
-def mocked_run_scans(wanted_scans: set[str], errored_scans: set[str] = set()):
+def mocked_run_scans(
+    wanted_scans: set[str], errored_scans: set[str] = set(), dp: Optional[DataProvider] = None
```

Comment:
If you don't want to import typing, I think this is also valid:

```suggestion
    wanted_scans: set[str], errored_scans: set[str] = set(), dp: DataProvider | None = None
```

Reply:
Done in cdbb3a18a3e2e66699dc224240fa71e91249c2d6